orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <queue>
+#include <string>
+
+#include "core/optimizer/graph_transformer.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/compute_optimizer/shared_utils.h"
+
+namespace onnxruntime {
+
+/**
+ * @Class PaddingElimination
+ *
+ * @brief Graph transformer that eliminates unnecessary padding computation caused by embedding sparsity.
+ * This transformer is implemented in the following steps:
+ * 1. Iterate the graph and find the Embedding node that matches these requirements:
+ *     (1) Its 2nd input is a graph input and its rank > 2 with the first two dimensions are dim_params which are
+ *         actually batch_size and sequence_length.
+ * Note: Now only support the case of the first two dimensions to merged and remove the padding on the merged
```

Comment:
shall we add support for the concrete dims, to support Ads models?

Reply:
Yes, I will do it after this pr.